### PR TITLE
Fix code and doc

### PR DESCRIPTION
**Pull request by mbaudin47**
This PR includes various updates.
- Indent CMakeLists.txt
- Removes mention and use of DistributionCollection in a script and a API help page.
- Simplifies the implementation of `TensorProductExperiment`after a suggestion from #1963

### CODE REVIEW

**Review by josephmure — APPROVED** [review]

**Review by josephmure — APPROVED** [review]
PR twice approved on my part!

**Review by sofianehaddad — APPROVED** [review]

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py [hunk 1]
```diff
@@ -347,9 +347,9 @@ def printCovarianceParameterChange(covarianceModel1, covarianceModel2):
 # First, we create a multivariate distribution, based on independent `Uniform` marginals which have the bounds required by the covariance model.
 
 # %%
-distributions = ot.DistributionCollection()
+distributions = []
 for i in range(dim):
-    distributions.add(ot.Uniform(lbounds[i], ubounds[i]))
+    distributions.append(ot.Uniform(lbounds[i], ubounds[i]))
 boundedDistribution = ot.ComposedDistribution(distributions)
```
Comment (regislebrun) [inline python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py:353]:
```
distributions = [ot.Uniform(lbounds[i], ubounds[i]) for i in range(dim)]
```

**Review by regislebrun — APPROVED** [review]

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py [hunk 2]
```diff
@@ -347,9 +347,9 @@ def printCovarianceParameterChange(covarianceModel1, covarianceModel2):
 # First, we create a multivariate distribution, based on independent `Uniform` marginals which have the bounds required by the covariance model.
 
 # %%
-distributions = ot.DistributionCollection()
+distributions = []
```
Comment (jschueller) [inline python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py:350]:
```suggestion
distributions = [ot.Uniform(lbounds[i], ubounds[i]) for i in range(dim)]
```

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py [hunk 3]
```diff
@@ -347,9 +347,9 @@ def printCovarianceParameterChange(covarianceModel1, covarianceModel2):
...
 for i in range(dim):
```
Comment (jschueller) [inline python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py:351]:
```suggestion
```

File: python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py [hunk 4]
```diff
@@ -347,9 +347,9 @@ def printCovarianceParameterChange(covarianceModel1, covarianceModel2):
...
-    distributions.add(ot.Uniform(lbounds[i], ubounds[i]))
+    distributions.append(ot.Uniform(lbounds[i], ubounds[i]))
```
Comment (jschueller) [inline python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_hyperparameters_optimization.py:352]:
```suggestion
```

**Review by regislebrun — APPROVED** [review]

**Pull request merged**